src/redux/sagas/motions/domains/createEditDomainMultiSigMotion.ts

```diff
@@ -0,0 +1,221 @@
+import { ClientType, Id } from '@colony/colony-js';
+import { AddressZero } from '@ethersproject/constants';
+import { call, fork, put, takeEvery } from 'redux-saga/effects';
+
+import { PERMISSIONS_NEEDED_FOR_ACTION } from '~constants/actions.ts';
+import { ContextModule, getContext } from '~context/index.ts';
+import { TRANSACTION_METHODS } from '~types/transactions.ts';
+
+import { ActionTypes } from '../../../actionTypes.ts';
+import { type AllActions, type Action } from '../../../types/actions/index.ts';
+import {
+  createTransaction,
+  createTransactionChannels,
+  getTxChannel,
+  waitForTxResult,
+} from '../../transactions/index.ts';
+import {
+  putError,
+  takeFrom,
+  // uploadIfpsAnnotation,
+  getColonyManager,
+  uploadAnnotation,
+  initiateTransaction,
+  getPermissionProofsLocal,
+} from '../../utils/index.ts';
+
+import { handleDomainMetadata } from './utils/handleDomainMetadata.ts';
+
+// import { ipfsUpload } from '../ipfs';
+
+function* createEditDomainMultiSigMotion({
+  payload: {
+    colonyAddress,
+    colonyName,
+    domainName,
+    domainColor,
+    domainPurpose,
+    annotationMessage,
+    domain,
+    isCreateDomain,
+    parentDomainId = Id.RootDomain,
+    customActionTitle,
+    colonyDomains,
+    colonyRoles,
+    domainCreatedInNativeId,
+  },
+  meta: { id: metaId, navigate, setTxHash },
+  meta,
+}: Action<ActionTypes.MOTION_MULTISIG_DOMAIN_CREATE_EDIT>) {
+  let txChannel;
+
+  try {
+    const apolloClient = getContext(ContextModule.ApolloClient);
+
+    /*
+     * Validate the required values
+     */
+    if (!domainName) {
+      throw new Error('A domain name is required to create a new domain');
+    }
+
+    if (!isCreateDomain && !domain?.id) {
+      throw new Error('A domain id is required to edit domain');
+    }
+
+    txChannel = yield call(getTxChannel, metaId);
+
+    /* additional editDomain check is for the TS to not ring alarm in getPermissionProofs */
+    const domainNativeId =
+      !isCreateDomain && domain?.nativeId ? domain.nativeId : parentDomainId;
+
+    // setup batch ids and channels
+    const batchKey = TRANSACTION_METHODS.CreateMotion;
+
+    const { createMotion, annotateMotion } = yield createTransactionChannels(
+      metaId,
+      ['createMotion', 'annotateMotion'],
+    );
+
+    /*
+     * Upload domain metadata to IPFS
+     */
+    // let domainMetadataIpfsHash = null;
+    // domainMetadataIpfsHash = yield call(
+    //   ipfsUpload,
+    //   JSON.stringify({
+    //     domainName,
+    //     domainColor,
+    //     domainPurpose,
+    //   }),
+    // );
```

Comment:
This can go (as well as the related imports)

Reply:
I honestly thought this was being tracked by someone because it has been in the codebase for so long! 😂 @iamsamgibbs

Reply:
Ignore me 😂